src/main/java/com/github/manolo8/darkbot/gui/titlebar/BackpageTask.java

```diff
@@ -0,0 +1,264 @@
+            Popups.of("Backpage exception", message)
+                    .messageType(JOptionPane.ERROR_MESSAGE)
+                    .showAsync();
+        } finally {
+            button.removeProgressBar();
+        }
+
+        return false;
+    }
+
+    private boolean checkVersion(Version current) throws IOException {
+        JProgressBar progressBar = button.createProgressBar();
+
+        ReleaseInfo releaseInfo = ReleaseInfo.get();
+        if (releaseInfo == null) return false;
+
+        Version remoteVersion = releaseInfo.getVersion();
+        ReleaseInfo.Asset asset = releaseInfo.getValidAsset();
+        if (remoteVersion == null || asset == null) return false;
+
+        VERSION_CHECK_TIMER.activate(); // activate the timer here - version will be successfully checked
+        return current != null && current.compareTo(remoteVersion) >= 0
+                || askUserToDownload(progressBar, current, remoteVersion, asset);
+    }
+
+    //return true if is possible to run a backpage
+    private boolean askUserToDownload(JProgressBar progressBar,
+                                      Version current, Version remote, ReleaseInfo.Asset asset) throws IOException {
+        String message = current == null ? I18n.get("gui.backpage_button.download_message")
+                : I18n.get("gui.backpage_button.new_version_message");
+        message += "\n -Ver: " + (current != null ? current + " -> " : "") + remote + ", size: " + (asset.size >> 20) + "MB";
+
+        JButton download = new JButton(I18n.get("gui.backpage_button.download"));
+        Popups.Builder builder = Popups.of("Backpage browser", message)
+                .messageType(JOptionPane.QUESTION_MESSAGE)
+                .options(download, I18n.get("gui.backpage_button.cancel"));
+
+        JOptionPane optionPane = builder.build();
+        download.addActionListener(l -> optionPane.setValue(download));
+
+        if (download == builder.showSync()) {
+            if (!clearDirectory())
+                throw new DirectoryClearException(BACKPAGE_PATH);
+
+            FileUtils.createDirectories(BACKPAGE_PATH);
+            downloadBackpage(progressBar, asset);
+            writeVersionFile(remote);
+        }
+
+        return readVersionFile() != null;
+    }
+
+    // true if backpage folder is clear
+    private boolean clearDirectory() throws IOException {
+        if (Files.notExists(BACKPAGE_PATH)) return true;
+
+        try (Stream<Path> walk = Files.walk(BACKPAGE_PATH)) {
+            boolean backapgeFileAccessible = walk.sorted(Comparator.reverseOrder())
+                    .filter(path -> path.toString().contains(EXECUTABLE_NAME))
+                    .map(Path::toFile)
+                    .map(File::delete)
+                    .findFirst().orElse(true);
+
+            if (!backapgeFileAccessible) return false; // probably used by another process - cannot delete
```

Comment:
```suggestion
            // Failed to delete, likely being used by another process
            if (!backapgeFileAccessible) return false;
```
Avoid trailing comments when possible, prefer placing them above the statement. This comment in particular is probably the only one that explains something non-obvious, make sure to keep it